### ISSUE TITLE
If there are quite a few artists in one album they overflow the UI.

### ISSUE THREAD

**Issue by fwuffyboi** [jellyfin/jellyfin-web#4228]
**Describe The Bug**
When an album i have in my music library has quite a few artists (about 15) they overflow off the UI bar where the album buttons are. (Heart, Shuffle, Mix, Play)

**Expected Behavior**
I don't expect anything really, besides for it to not overflow lol.
maybe if there are over 10 artists: say something along the lines of "and {} others..." after it says the first 10? 
A quick and dirty inspect element job kinda shows what i mean: 
![image](https://user-images.githubusercontent.com/73500040/209684135-67352e31-e8c9-418f-8075-10ec3c0bcc34.png)

**Screenshots**
![image](https://user-images.githubusercontent.com/73500040/209682567-70c069e9-54ac-4bc2-95bc-aa513d803ca1.png)

**System (please complete the following information):**
 - Platform: Linux, ChromeOS
 - Browser: Chrome, Firefox
 - Jellyfin Version: 10.8.7

**Comment by danditomaso** [comment]
I came over from the Jellyfin sub Reddit, after learning that the best way to help 8s through OSS contributions. I've been a React dev for the past 4 years and would like to take this issue on

**Comment by nielsvanvelzen** [comment]
Feel free to work on this issue. As you might have found out already; not all parts of the web client are rewritten in React. I'm not sure about this specific page (I don't really work on the web client). If you have any questions our developers are most active on [Matrix/Discord](jellyfin.org/contact/).

**Comment by acreiko** [comment]
Hi, just wanted to ask out of curiosity if this issue has been resolved or worked on yet. thanks.

**Comment by thornbill** [comment]
No... the status of this issue should be updated if anything changes

**Comment by fwuffyboi** [comment]
bump. this still affects me on the latest docker version of jellyfin web.
![image](https://github.com/jellyfin/jellyfin-web/assets/73500040/00e3e9dc-a961-4df4-9580-954f27e574dc)

**Comment by alfred-delacosta** [comment]
I would like to fix this issue. I will be working on and when I finish I will submit my PR.

**Comment by thornbill** [comment]
Could someone verify the fix in #4830 is working as expected by testing the Preview URL: https://github.com/jellyfin/jellyfin-web/pull/4830#issuecomment-1777917418

I can't seem to find an album that has enough artists to trigger this bug in my library. :sweat_smile:

**Comment by felix920506** [comment]
Image from @Xitee1 (https://github.com/jellyfin/jellyfin/issues/10052#issuecomment-1868077418) to show extreme cases
![圖片](https://github.com/jellyfin/jellyfin-web/assets/25688628/0edbdd33-f764-46c6-b468-27951a148055)

**Comment by twihno** [comment]
As I already mentioned in jellyfin/jellyfin#10052, I don't think that concatinating the artists is a good idea.
The problem with these kinds of albums (at least in my opinion) is that they normally are some kind of compilation with no defined album artist. The way e.g. spotify solves this issue is to show "Various artists" (unclickable) as the album artist.

Concatinating the available artists may fix the overflow, but regarding UX you still have another problem:
Jellyfin web doesn't show the artist of the individual tracks if it "matches" with the album artist.
Let's say you have an album with multiple remixes or

> bump. this still affects me on the latest docker version of jellyfin web. ![image](https://private-user-images.githubusercontent.com/73500040/255041708-00e3e9dc-a961-4df4-9580-954f27e574dc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY5MTk2NjgsIm5iZiI6MTcwNjkxOTM2OCwicGF0aCI6Ii83MzUwMDA0MC8yNTUwNDE3MDgtMDBlM2U5ZGMtYTk2MS00ZGY0LTk1ODAtOTU0ZjI3ZTU3NGRjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjAzVDAwMTYwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIxYmQ4MjY1NDhiMDc0YzVlMTJiMjZkZWUyZWU1NWJhYWY1NjRjNTAxMDE1NWY2NWU4YWQzNjIwYTUyZTQ1NjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.kPUEV6xOw2DL3KbWD7VkpHuOTa-ZCYEW1QaLH0-pD5I)

You don't get to know the artist corresponding to the track and the artists in the header are always sorted alphabetically which also doesn't help in this case.

What i proposed in the mentioned issue (jellyfin/jellyfin#10052) was:

> ## What should Jellyfin do?
>
> (My suggestions)
>
> - Maybe tag these albums (no album artist, multiple artists) as compilations or something like that? Perhaps clients can handle this specific case more easily with this information
> - Show "Various Artists" (best case: translated) as album artist without generating an artist page. Clients should not make this text clickable. Maybe an option would be to leave the album artist field empty (I don't know what the current behavior is) and let the clients handle this case. Hot take: every piece of music/media was created by at least one person. So if no person is credited for the whole unit/album showing "Various Artists" would always be correct because multiple artists worked on it?
    *(I think this would be the best and easiest option)*
> - Jellyfin-web: Show the individual artist per Track

That were just the ideas I came up with but I think that shortening the list isn't a good solution to this problem

**Comment by twihno** [comment]
I tried investigating further and noticed some interesting behavior: https://github.com/jellyfin/jellyfin/issues/10052#issuecomment-1925000577

TLDR: An empty album artist leads to a bad import which messes with the UI. The UI also doesn't know how to handle an empty album artist
